### PR TITLE
Uses RecordCursor in RecordStore.Scanner

### PR DESCRIPTION
**Pull request by tinwelint**
instead of doing getRecord each time, benefit is one open PageCursor
and not repinning page for every record.

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/impl/store/RecordStore.java [hunk 1]
```diff
@@ -458,37 +451,45 @@ public abstract void processRelationshipGroup( RecordStore<RelationshipGroupReco
...
         {
-            return () -> new PrefetchingIterator<R>()
+            return () -> new PrefetchingResourceIterator<R>()
             {
                 final PrimitiveLongIterator ids = new StoreIdIterator( store, forward );
-                final R record = store.newRecord();
+                final RecordCursor<R> cursor = store.newRecordCursor( store.newRecord() );
+                {
+                    store.placeRecordCursor( 0, cursor, RecordLoad.CHECK );
+                }
```
Comment (chrisvest) [inline community/kernel/src/main/java/org/neo4j/kernel/impl/store/RecordStore.java]: I think the code will be clearer if you extracted this inner class, and moved this initialisation to it's constructor.

**Comment by chrisvest** [comment]
@tinwelint Had one comment.

**Comment by tinwelint** [comment]
@chrisvest addressed that comment in another commit

**Comment by chrisvest** [comment]
Beautiful.

**Pull request merged**